core/linux.go

```diff
@@ -47,6 +47,9 @@ func pathToLibFlag(path string) string {
 	_, base := filepath.Split(path)
 	ext := filepath.Ext(base)
 	base = strings.TrimSuffix(base, ext)
+	if !strings.HasPrefix(base, "lib") {
+		panic(errors.New("Shared library need to start with 'lib' prefix"))
```

Comment:
Grammar: s/need/needs

Although maybe "Shared library output must start with 'lib' prefix" would be clearer?

Reply:
Actually it's not only referring to output field but name in general, so I've changed it as you suggested but with minor change